server/app/controllers/applicant/ApplicantProgramBlocksController.java

```diff
@@ -1221,15 +1224,21 @@ private CompletionStage<Result> renderIneligiblePage(
           () -> ok(northStarApplicantIneligibleView.render(params)).as(Http.MimeTypes.HTML));
     } else {
       return supplyAsync(
-          () ->
-              ok(
+          () -> {
+            try {
```

Comment:
**question:** I don't think this needs to be wrapped in a try / catch, since we're just passing through the blockId passed into this function?

Reply:
It shows error if we don't handle ProgramBlockDefinitionNotFoundException, as block was not passed into the function previously. So I added try and catch to handle it. Let me know if there's another way you think is better for this tho.